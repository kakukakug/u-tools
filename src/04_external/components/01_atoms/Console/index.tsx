import React, { useState } from "react";
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  TextInput,
} from "react-native";

import { Colors } from "../../../styles/Colors";

const styles = StyleSheet.create({
  consoleContainer: {
    width: "100%",
    backgroundColor: Colors.text,
  },
  button: {
    width: "100%",
    padding: 10,
    alignItems: "center",
  },
  buttonText: {
    fontSize: 18,
    paddingVertical: 5,
    color: Colors.white,
    textAlign: "center",
  },
  consoleInput: {
    color: Colors.background,
    fontSize: 16,
    height: 250,
    marginHorizontal: 10,
    marginTop: 10,
    padding: 5,
    backgroundColor: Colors.icon,
    borderRadius: 4,
    textAlignVertical: "top",
  },
});

type OuterProps = {
  consoleText: string;
};

export const Console = (props: OuterProps) => {
  const [isVisible, setIsVisible] = useState(true);
  const { consoleText } = props;

  if (isVisible === false) {
    return (
      <View style={styles.consoleContainer}>
        <TouchableOpacity
          style={styles.button}
          onPress={() => {
            setIsVisible(true);
          }}>
          <Text style={styles.buttonText}>open output text</Text>
        </TouchableOpacity>
      </View>
    );
  }
  return (
    <View style={styles.consoleContainer}>
      <TextInput
        style={styles.consoleInput}
        multiline={true}
        editable={false}
        value={consoleText}
        testID={"console-text"}
      />
      <TouchableOpacity
        style={styles.button}
        onPress={() => {
          setIsVisible(false);
        }}>
        <Text style={styles.buttonText}>close</Text>
      </TouchableOpacity>
    </View>
  );
};
