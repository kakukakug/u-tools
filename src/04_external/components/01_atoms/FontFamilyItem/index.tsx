import React from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";
import { Colors } from "../../../styles/Colors";

const styles = StyleSheet.create({
  familyButton: {
    backgroundColor: Colors.surface,
    padding: 10,
    marginVertical: 4,
  },
  familyText: {
    color: Colors.text,
  },
});

export const FontFamilyItem = (props) => {
  const { name, onPress, selectFamily } = props;
  return (
    <TouchableOpacity
      onPress={() => {
        onPress(name);
      }}
      style={[
        styles.familyButton,
        name === selectFamily
          ? { backgroundColor: Colors.primary }
          : { backgroundColor: Colors.surface },
      ]}
      testID={"fontFamilyItem"}>
      <Text style={styles.familyText}>{name}</Text>
    </TouchableOpacity>
  );
};
