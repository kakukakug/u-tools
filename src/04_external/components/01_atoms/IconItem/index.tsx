import React from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";
import {
  AntDesign,
  Entypo,
  EvilIcons,
  Feather,
  FontAwesome,
  FontAwesome5,
  Foundation,
  Ionicons,
  MaterialIcons,
  MaterialCommunityIcons,
  SimpleLineIcons,
  Octicons,
  Zocial,
  Fontisto,
} from "@expo/vector-icons";

import { Colors } from "../../../styles/Colors";

const styles = StyleSheet.create({
  iconButton: {
    width: 120,
    margin: 4,
    padding: 10,
    backgroundColor: Colors.white,
    borderRadius: 10,
    justifyContent: "center",
    alignItems: "center",
  },
  iconName: {
    fontSize: 14,
    margin: 4,
    color: Colors.text,
  },
});

type OuterProps = {
  name: string;
  onPress: (props: string) => void;
  selectFamily: string;
};

const iconSize = 42;

export const IconItem = (props: OuterProps) => {
  const { name, onPress, selectFamily } = props;
  console.log("render icon")
  const getIconConponrnt = () => {
    switch (selectFamily) {
      case "AntDesign":
        return <AntDesign name={name} size={iconSize} color={Colors.icon} />;
      case "Octicons":
        return <Octicons name={name} size={iconSize} color={Colors.icon} />;
      case "Entypo":
        return <Entypo name={name} size={iconSize} color={Colors.icon} />;
      case "EvilIcons":
        return <EvilIcons name={name} size={iconSize} color={Colors.icon} />;
      case "Feather":
        return <Feather name={name} size={iconSize} color={Colors.icon} />;
      case "FontAwesome":
        return <FontAwesome name={name} size={iconSize} color={Colors.icon} />;
      case "FontAwesome5":
        return <FontAwesome5 name={name} size={iconSize} color={Colors.icon} />;
      case "Foundation":
        return <Foundation name={name} size={iconSize} color={Colors.icon} />;
      case "Ionicons":
        return <Ionicons name={name} size={iconSize} color={Colors.icon} />;
      case "MaterialIcons":
        return (
          <MaterialIcons name={name} size={iconSize} color={Colors.icon} />
        );
      case "MaterialCommunityIcons":
        return (
          <MaterialCommunityIcons
            name={name}
            size={iconSize}
            color={Colors.icon}
          />
        );
      case "SimpleLineIcons":
        return (
          <SimpleLineIcons name={name} size={iconSize} color={Colors.icon} />
        );
      case "Zocial":
        return <Zocial name={name} size={iconSize} color={Colors.icon} />;
      case "Fontisto":
        return <Fontisto name={name} size={iconSize} color={Colors.icon} />;
      default:
        return <></>;
    }
  };
  return (
    <TouchableOpacity
      onPress={() => {
        onPress(name);
      }}
      style={styles.iconButton}>
      {getIconConponrnt()}
      <Text style={styles.iconName}>{name}</Text>
    </TouchableOpacity>
  );
};
