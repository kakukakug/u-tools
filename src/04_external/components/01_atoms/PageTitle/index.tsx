import React from "react";
import { StyleSheet, Text, View } from "react-native";

import { Colors } from "../../../styles/Colors";

const styles = StyleSheet.create({
  title: { marginVertical: 10, padding: 10 },
  titleText: { fontSize: 38, color: Colors.text },
});

type OuterProps = {
  title: string;
};
export const PageTitle = (props: OuterProps) => {
  const { title } = props;

  return (
    <View style={styles.title}>
      <Text style={styles.titleText}>{title}</Text>
    </View>
  );
};
