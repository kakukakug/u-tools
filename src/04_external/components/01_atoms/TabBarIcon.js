import React from "react";
import { MaterialCommunityIcons } from "@expo/vector-icons";

import { Colors } from "../../styles/Colors";

export default function TabBarIcon(props) {
  return (
    <MaterialCommunityIcons
      name={props.name}
      size={26}
      color={props.focused ? Colors.tabIconSelected : Colors.tabIconDefault}
    />
  );
}
