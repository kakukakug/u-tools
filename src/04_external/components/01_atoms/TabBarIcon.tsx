import React from "react";
import { MaterialCommunityIcons } from "@expo/vector-icons";

import { Colors } from "src/04_external/styles/Colors";

type OuterProps = {
  name: string;
  focused: boolean;
};

export const TabBarIcon = (props: OuterProps) => {
  const { name, focused } = props;
  return (
    <MaterialCommunityIcons
      name={name}
      size={26}
      color={focused ? Colors.tabIconSelected : Colors.tabIconDefault}
    />
  );
};
