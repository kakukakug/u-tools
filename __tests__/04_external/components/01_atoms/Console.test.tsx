import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { Console } from "../../../../src/04_external/components/01_atoms/Console";

describe("Console", () => {
  const mockConsoleText = "test console";
  it(`Consoleを描写`, () => {
    const component = render(<Console consoleText={mockConsoleText} />);

    const output = component.getByDisplayValue(mockConsoleText);
    expect(output).toBeTruthy();

    expect(component.toJSON()).toMatchSnapshot();
  });
  it(`button をタップしたとき、consoleが閉じる`, async () => {
    const component = render(<Console consoleText={mockConsoleText} />);

    const output = component.getByDisplayValue(mockConsoleText);
    expect(output).toBeTruthy();

    const closeButton = component.getByText("close");
    fireEvent.press(closeButton);

    const noOutput = component.queryByDisplayValue(mockConsoleText);
    expect(noOutput).toBeNull();

    const openButton = component.getByText("open output text");
    expect(openButton).toBeTruthy();

    expect(component.toJSON()).toMatchSnapshot();
  });
});
