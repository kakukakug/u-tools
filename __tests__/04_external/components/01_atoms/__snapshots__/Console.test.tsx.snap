// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Console Consoleを描写 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#333",
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    editable={false}
    multiline={true}
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#666",
        "borderRadius": 4,
        "color": "#fffffe",
        "fontSize": 16,
        "height": 250,
        "marginHorizontal": 10,
        "marginTop": 10,
        "padding": 5,
        "textAlignVertical": "top",
      }
    }
    testID="console-text"
    underlineColorAndroid="transparent"
    value="test console"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fffffe",
          "fontSize": 18,
          "paddingVertical": 5,
          "textAlign": "center",
        }
      }
    >
      close
    </Text>
  </View>
</View>
`;

exports[`Console button をタップしたとき、consoleが閉じる 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#333",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fffffe",
          "fontSize": 18,
          "paddingVertical": 5,
          "textAlign": "center",
        }
      }
    >
      open output text
    </Text>
  </View>
</View>
`;
